CoreAnalyticView/src/au/gov/asd/tac/constellation/views/analyticview/AnalyticConfigurationPane.java

```diff
@@ -582,4 +620,71 @@ public String getName() {
             return "Analytic View: Update State";
         }
     }
+    
+    /**
+     * Update the display by reading and writing to/from the state attribute. 
+     */
+    private static final class AnalyticViewStateUpdater extends SimpleEditPlugin {
+
+        private final AnalyticConfigurationPane analyticConfigurationPane;
+        private final boolean pluginWasSelected;
+        
+        public AnalyticViewStateUpdater(final AnalyticConfigurationPane analyticConfigurationPane, final boolean categoryWasSelected) {
+            this.analyticConfigurationPane = analyticConfigurationPane;
+            this.pluginWasSelected = categoryWasSelected;
+        }
+
+        @Override
+        public void edit(final GraphWriteMethods graph, final PluginInteraction interaction, final PluginParameters parameters) throws InterruptedException, PluginException {
+            String currentCategory = analyticConfigurationPane.categoryList.getSelectionModel().getSelectedItem();
+            int stateAttributeId = AnalyticViewConcept.MetaAttribute.ANALYTIC_VIEW_STATE.ensure(graph);
+            
+            // Make a copy in case the state on the graph is currently being modified.
+            AnalyticViewState currentState = graph.getObjectValue(stateAttributeId, 0);
+            currentState = (currentState == null) ? new AnalyticViewState() : new AnalyticViewState(currentState);
+            
+            if (pluginWasSelected) {
+                // remove all plugins matching category
+                currentState.removePluginsMatchingCategory(currentCategory);
+                // grab all plugins from currently selected category
+                List<SelectableAnalyticPlugin> checkedPlugins = new ArrayList<>();
+                // adding items to checkedPlugins array when they are selected
+                analyticConfigurationPane.pluginList.getItems().forEach(selectablePlugin -> {
+                    if (selectablePlugin.isSelected()) {
+                        checkedPlugins.add(selectablePlugin);
+                    }
+                });
+                if (!checkedPlugins.isEmpty()) {
+                    currentState.addAnalyticQuestion(analyticConfigurationPane.currentQuestion, checkedPlugins);
+                }
+                graph.setObjectValue(stateAttributeId, 0, currentState);
+            }
+            
+            // get question only if one exists 
```

Comment:
Shouldn't all this be in the state listener on the top component?

Reply:
_I can take a look at refactoring the code to include these state listeners if you feel they would benefit as opposed to the current solution._